OIDC: Add a property enabling to use client authentication for discovery and JWKS endpoints (#54433)
quarkusio/quarkus: docs/src/main/asciidoc/security-oidc-expanded-configuration.adoc

|====
|Property name |Default |Description

> |quarkus.oidc.credentials.for-all-endpoints |false|Send credentials to all OIDC endpoints including discovery and JWKS. See xref:security-oidc-code-flow-authentication.adoc#authenticated-access-to-all-oidc-endpoints[Authenticated access to all OIDC endpoints].

Reviewer: ```suggestion
|quarkus.oidc.credentials.for-all-endpoints |false|Send credentials to all OIDC endpoints including the discovery and JWKS endpoints..
```

In this doc, the properties are usually explained below, and indeed, like you did, so propose to shorten it here

|====

Set `quarkus.oidc.credentials.jwt.source=bearer` if the client authentication token is provided by Kubernetes, and use `quarkus.oidc.credentials.jwt.token-path` to point to the file resource containing this token.

> If the OIDC provider, such as Kubernetes, also enforces authenticated access to the discovery and JWKS endpoints, set `quarkus.oidc.credentials.for-all-endpoints=true`.

Reviewer: Let's try to align with other docs, but also please update the table that lists various credential properties

(Also, this is offtopic for this PR - the line above mentions a bearer source - but we also now support spiffe - I'm typing it just not to forget, we can try another tiny PR so that it can be backported)

Author: > Let's try to align with other docs, but also please update the table that lists various credential properties

sure

> (Also, this is offtopic for this PR - the line above mentions a bearer source - but we also now support spiffe - I'm typing it just not to forget, we can try another tiny PR so that it can be backported)

I tried to look it out before I made this PRand SPIFFE I think is not related as I didn't find a use case, so let's do whatever is needed as the follow-up.

Author: I have moved it into the client secret table, not sure which other table you meant.

|====

Set `quarkus.oidc.credentials.jwt.source=bearer` if the client authentication token is provided by Kubernetes, and use `quarkus.oidc.credentials.jwt.token-path` to point to the file resource containing this token.

> If the OIDC provider also enforces authenticated access to the discovery and JWKS endpoints, see xref:security-oidc-code-flow-authentication.adoc#authenticated-access-to-all-oidc-endpoints[Authenticated access to all OIDC endpoints].

Reviewer: In this doc I try to briefly explain how to use a property, so I'd suggest 

```suggestion
Set `quarkus.oidc.credentials.for-all-endpoints=true` when the OIDC provider enforces authenticated access to the discovery and JWKS endpoints. See xref:security-oidc-code-flow-authentication.adoc#authenticated-access-to-all-oidc-endpoints[Authenticated access to all OIDC endpoints] for more information.
```

Author: applied, thx
